### ISSUE TITLE
Missing xdg-desktop-portal-gtk

### ISSUE THREAD

**Issue by dos1** [ValveSoftware/SteamOS#803]
xdg-desktop-portal-gtk is required under Plasma Wayland session for flatpaked GTK apps to be able to read GTK settings, including font antialiasing. Not having this portal installed causes them to end up with bad font rendering.

![20220703003606_1](https://user-images.githubusercontent.com/126204/177017977-1f0503ec-e5d3-4537-8562-31449f3f3b88.jpg)

**Comment by bertogg** [comment]
I think you're right, upstream bug report here: https://github.com/flatpak/flatpak/issues/2861#issuecomment-494145504

**Comment by bertogg** [comment]
I had a moment to check this, the problem is as you say but the Desktop mode defaults to X11, are you launching the Wayland session manually?

**Comment by dos1** [comment]
Yes, just as mentioned in the description - this is a great way to quickly get into desktop without ending gamescope session and retaining access to input configuration overlay.

**Comment by different55** [comment]
Think this might be a duplicate of #775? Both are problems caused by missing the proper desktop portal being installed.

**Comment by bertogg** [comment]
No, I don't think it's the same problem.

#775 is about the portal not working properly after restarting the desktop session. Without the fix, this problem can be reproduced easily in a standard configuration.

#803 (this one) one is about gsettings not working properly because the portal that implements that functionality is not installed at all. This only happens with a non-standard configuration (Plasma with Wayland) so it's not so severe in my opinion.

**Comment by popsUlfr** [comment]
I'm also mostly using "desktop mode" from the Steam Deck UI this way as it's just starts faster and is more convenient with the quick access menu and all at my fingertips. I can still launch games at the same time through the Steam Deck UI.

I made my non-steam game shortcut that looks like this:
```
env -u LD_PRELOAD startplasma-wayland --xwayland --x11-display "$DISPLAY" --wayland-display "$GAMESCOPE_WAYLAND_DISPLAY" --width 1280 --height 800 --no-lockscreen -- plasma_session
```

GTK flatpak apps have non-antialiased fonts and missing icons at times (like bottles) so It would be great if `xdg-desktop-portal-gtk` and `xdg-desktop-portal-gnome` (mostly fixes the missing icons thing) could be installed in the base system by default.

**Comment by dos1** [comment]
@popsUlfr Unlike xdg-desktop-portal-gtk, a correctly functioning Plasma system shouldn't need xdg-desktop-portal-gnome for anything, so the missing icon stuff points to something else being misconfigured.

**Comment by popsUlfr** [comment]
Yes, I agree that the gnome desktop portal shouldn't be needed if we don't use the gnome desktop environment anyway. But currently it seems that only by having both installed, apps like bottles will show up properly or rather as they were intended to look like by by the devs.

**Comment by cybik** [comment]
Adding to this thread: while not the main intended use for this platform, it appears that another flatpak, namely Zoom, also complains about the lack of `xdg-desktop-portal`, nevermind the specific `xdg-desktop-portal-gtk`. Right now, without this tool (allegedly), I can't use any SSO solution (custom, Google or whatever) to log into Zoom (flat) from the SteamOS desktop.

**Comment by evelikov** [comment]
@cybik OS 3.4 includes `xdg-desktop-portal` and Zoom SSO just works even without `xdg-desktop-portal-gtk`. Just played around with it the Google one.

**Comment by cybik** [comment]
Hm. That's quaint then. I can validate the portal layer is indeed present on my system - so apologies for necro'ing this post. Something else might be afoot.

Full disclosure, I'm trying to not install anything in the core system; my browser is actually a Flatpak-installed Edge.

**Comment by bertogg** [comment]
FYI the GTK portal will be preinstalled in SteamOS 3.6.
